packages/apps/types/src/schema/thread.ts

```diff
@@ -4,104 +4,63 @@
-  ARCHIVED = 1,
-  DELETED = 2, // TODO(burdon): Actually delete (need sync range so that doesn't return).
-  SPAM = 3,
-}
-
-export class ContactType extends TypedObject({ typename: 'dxos.org/type/Contact', version: '0.1.0' })({
-  name: S.optional(S.String),
-  identifiers: S.mutable(
-    S.Array(
-      S.Struct({
-        type: S.String,
-        value: S.String,
-      }),
-    ),
-  ),
-}) {}
-
-const _RecipientSchema = S.mutable(
+export const EntitySchema = S.mutable(
   S.Struct({
+    contact: S.optional(ref(ContactType)),
+    // TODO(wittjosiah): Should the below fields just be the contact schema?
+    //   i.e. it should either be a reference to an existing contact or an inline contact schema.
     identityKey: S.optional(S.String),
     email: S.optional(S.String),
     name: S.optional(S.String),
-    contact: S.optional(ref(ContactType)),
   }),
 );
 
-export interface RecipientType extends S.Schema.Type<typeof _RecipientSchema> {}
-
-const _BlockSchema = S.Struct({
+export class MessageType extends TypedObject({ typename: 'dxos.org/type/Message', version: '0.1.0' })({
+  /** ISO date string when the message was sent. */
   timestamp: S.String,
-  // TODO(burdon): Should not be a separate object.
-  content: S.optional(ref(TextV0Type)),
-  object: S.optional(ref(Expando)),
-});
-
-export interface BlockType extends S.Schema.Type<typeof _BlockSchema> {}
-
-// TODO(wittjosiah): This is way too coupled with email.
-//  Thread messages and email messages should not be the same thing.
-//  Interoperability should be handled by lenses or some other transformation mechanism.
-//  Requiring the same schema for all types of messages will not scale -
-//  It also makes the simple cases much more complex than they need to be.
-//  [burdon]: The intent of course is to enable a universal inbox out of the box.
-export class MessageType extends TypedObject({ typename: 'braneframe.Message', version: '0.1.0' })({
-  type: S.optional(S.String),
-  date: S.optional(S.String),
-  from: _RecipientSchema,
-  to: S.optional(S.Array(_RecipientSchema)),
-  cc: S.optional(S.Array(_RecipientSchema)),
-  subject: S.optional(S.String),
-  blocks: S.mutable(S.Array(_BlockSchema)),
-  links: S.optional(S.Array(ref(Expando))),
-  state: S.optional(S.Enums(MessageState)),
-  read: S.optional(S.Boolean),
-  context: S.optional(
-    S.Struct({
-      space: S.optional(S.String),
-      schema: S.optional(S.String),
-      object: S.optional(S.String),
-    }),
-  ),
-}) {}
-
-export class ThreadType extends TypedObject({ typename: 'braneframe.Thread', version: '0.1.0' })({
-  title: S.optional(S.String),
-  messages: S.mutable(S.Array(ref(MessageType))),
-  // TODO(burdon): Reconcile with Message.Context.
+  /** Identity of the message sender. */
+  sender: EntitySchema,
+  /** Text content of the message. */
+  content: S.String,
+  /** Non-text content embedded into a message (e.g. files, polls, etc.) */
+  embeds: S.optional(S.Array(ref(Expando))),
```

Comment:
parts? it would be ideal to be tuples of types + generic references (i.e., not necessarily expandos).